### ISSUE TITLE
Refactor complex dependency related code

### ISSUE THREAD

**Issue by edubkendo** [rsense/rsense#35]
The code for loading dependencies is extremely complex and hairy. It should be possible to dramatically simplify it by just checking a projects Gemfile.lock with something like this:

``` ruby
lockfile = Bundler::LockfileParser.new(Bundler.read_file(Pathname.new("~/tmp/cool/Gemfile.lock").expand_path.to_s))
names = lockfile.dependencies.map {|d| d.name}

names.map do |n|
  { name: n, path: Bundler.rubygems.find_name(n).first.full_gem_path }
end
```

Should be able to remove huge globs of code in the following files:

https://github.com/rsense/rsense-server/blob/master/lib/rsense/server/command.rb
https://github.com/rsense/rsense-server/blob/master/lib/rsense/server/load_path.rb
https://github.com/rsense/rsense-server/blob/master/lib/rsense/server/gem_path.rb
https://github.com/rsense/rsense-server/blob/master/lib/rsense/server/command/preload.rb

Getting this done might even remove the problems we currently have with loading larger projects like Rails apps.

**Comment by nessamurmur** [comment]
Started looking at this but haven't quite found the best place for it yet and where this currently gets kicked off...

**Comment by edubkendo** [comment]
Yeah, I made a real mess of this ... First big ruby project done in too much of a hurry. :( 

The whole thing kicks off around here i think: https://github.com/rsense/rsense-server/blob/master/bin/_rsense.rb#L102
